File: 96. Unique Binary Search Trees.py
class Solution(object):
    def numTrees(self, n):
        """
        :type n: int
        :rtype: int
        """
        if n == 0 or n == 1:
            return 1
        T = [0 for i in xrange(n+1)]
        T[0], T[1] = 1, 1
        for i in xrange(2, n+1):
            for j in xrange(1, i+1):
                T[i] += T[j-1] * T[i-j]
        return T[n]

File: 96. Unique Binary Search Trees.cpp
class Solution {
public:
    int numTrees(int n) {
        if (n == 0 || n == 1) return 1;
        vector <int> T (n+1);
        T[0] = 1; T[1] = 1;
        for (int i = 2; i <= n; i++)
        {
            for (int j = 1; j <= i; j++)
                T[i] += T[j-1] * T[i-j];
        }
        return T[n];
    }
};
